src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -206,6 +206,17 @@ public Stream<SourceFile> listSourceFiles(MavenProject mavenProject, @Nullable X
                     .map(addProvenance(baseDir, projectProvenance, null));
             logDebug(mavenProject, "Parsed " + (alreadyParsed.size() - sourcesParsedBefore) + " additional files found within the project.");
             sourceFiles = Stream.concat(sourceFiles, parsedResourceFiles);
+        } else {
+            // Only parse Maven wrapper files, such that UpdateMavenWrapper can use the version information.
+            int sourcesParsedBefore = alreadyParsed.size();
+            Stream<SourceFile> parsedResourceFiles = Stream.concat(
+                            rp.parse(mavenProject.getBasedir().toPath().resolve(".mvn/wrapper"), alreadyParsed),
+                            Stream.concat(
+                                    rp.parse(mavenProject.getBasedir().toPath().resolve("mvnw"), alreadyParsed),
+                                    rp.parse(mavenProject.getBasedir().toPath().resolve("mvnw.bat"), alreadyParsed)))
```

Comment:
Don't ask me why, but on the Maven side they opted for `mvnw.cmd` instead of `mvnw.bat`.

Reply:
```suggestion
                                    rp.parse(mavenProject.getBasedir().toPath().resolve("mvnw.cmd"), alreadyParsed)))
```